.github/workflows/performance.yml

```diff
@@ -0,0 +1,120 @@
+
+# Schedule triggers
+on:
+  # TODO this is just while developing
+  pull_request:
+    branches:
+      - 'develop' 
+      - 'performance-regression-testing'
+  schedule:
+    # runs every day at 10:05pm
+    - cron:  '5 22 * * *'
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+
+  test-rust:
```

Comment:
What does the `test-rust` and `build-rust` steps do for us? I don't fully understand why they are needed for the perf testing. I know nothing about Cargo so it might be worth adding comments so it's more apparent

Reply:
added comments 👍